### PR TITLE
Start backport to Guzzle 6

### PR DESCRIPTION
**Pull request by bakura10**
ping @danizord

This is a start of work for moving this library to Guzzle 6. However as of today Guzzle 6 Services is not working at all (it's being updated).

I also moved the various signature verification and OAuth related methods to specific classes (happy ? ;)). Maybe we could move them already in current master as this is pretty big change but pretty important.

### CODE REVIEW

File: composer.json [hunk 1]
```diff
@@ -15,14 +15,20 @@
             "homepage": "http://www.michaelgallego.fr"
         }
     ],
+    "repositories": [
+        {
+            "type": "git",
+            "url": "https://github.com/guzzle/guzzle-services.git"
```
Comment (danizord) [inline composer.json]: Is not it on packagist yet?

File: index.php [hunk 2]
```diff
@@ -0,0 +1,12 @@
...
+require_once 'vendor/autoload.php';
+
+$sh = new \ZfrShopify\ShopifyClient([
+    'private_app' => true,
+    'api_key'     => 'a9ff874f7fb5f4bc3ea4dc2fee21442b',
+    'password'    => 'e8d5c744069062afd262ee946ed7ed19',
+    'shop'        => 'maestrooo-playground.myshopify.com'
+]);
+
+var_dump($sh->getShop());
```
Comment (danizord) [inline index.php]: Drop this file :D

File: src/OAuth/Authorizer.php [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+     *
+     * @param  string $apiKey
+     * @param  string $shopDomain
+     * @param  array  $scopes
+     * @param  string $redirectionUri
+     * @param  string $nonce
+     * @return ResponseInterface
+     * @throws Exception\MissingApiKeyException
+     */
+    public function createAuthorizationResponse(string $apiKey, string $shopDomain, array $scopes, string $redirectionUri, string $nonce)
```
Comment (danizord) [inline src/OAuth/Authorizer.php]: I think you can move it to it's own class:
```php
return new AuthorizationRedirectResponse($apiKey, $shopDomain, $scopes, $redirectionUri, $nonce);
```

File: composer.json (hunk 1 above)
Comment (bakura10) [inline composer.json]: There is only a branch :(. So wee need to do that.

File: index.php (hunk 2 above)
Comment (bakura10) [inline index.php]: Wow.... and it has the key, damn :p.

File: src/OAuth/Authorizer.php (hunk 3 above)
Comment (bakura10) [inline src/OAuth/Authorizer.php]: And how should I call the class that do the exchange? Those two operations are really linked together :)

**Comment by bakura10** [comment]
Replaced by #25

**Pull request closed without merging**